Geometry/CommonDetUnit/interface/GeomDet.h

```diff
@@ -119,9 +125,10 @@ class GeomDet {
 
   ReferenceCountingPointer<Plane>  thePlane;
   DetId m_detId;
-  int m_index;
+  int m_index=-1;
```

Comment:
It would be nice to cleanup all old code in general yes.
takes time and effort.
Let's schedule it for PhaseI (this code is candidate for backport to 800).